src/Middleware.php

```diff
@@ -197,7 +197,8 @@ function ($reason) use ($logger, $request, $formatter) {
                         $response = $reason instanceof RequestException
                             ? $reason->getResponse()
                             : null;
-                        $message = $formatter->format($request, $response, $reason);
+                        $exception = $reason instanceof RequestException ? $reason : null;
+                        $message = $formatter->format($request, $response, $exception);
```

Comment:
I have a question. Why `$reason` is string.

So if reason shouldn't be a string we could typehint like this:

`function (\Exception $reason) use ($logger, $request, $formatter) {`

if reason could be a string then I would suggest to improve this block of code with the anonymous function because we are checking twice about the exception

Also note that the formatter does not accept only `RequestException` but `Exceptions` in general.

Reply:
`$reason` is mixed (there is no typehint in the project) so theoretically it can be anything, but it is a string if we create a `RejectedPromise` with a string. example https://github.com/csarrazi/guzzle-cache-middleware/blob/master/src/MockMiddleware.php#L47

The formatter accept only null|Exception as third argument so we have to check the type of `$reason` before calling the formatter.

what optimisation do you suggest ?